src/main/java/com/aws/greengrass/util/S3SdkClientFactory.java

```diff
@@ -89,33 +91,54 @@ public S3Client getS3Client() throws DeviceConfigurationException {
      * @return s3client
      */
     public S3Client getClientForRegion(Region r) {
-        setS3EndpointType(Coerce.toString(deviceConfiguration.gets3EndpointType()));
+        handleSystemPropertyUpdate();
         return clientCache.computeIfAbsent(r, (region) -> S3Client.builder()
                 .httpClientBuilder(ProxyUtils.getSdkHttpClientBuilder())
                 .serviceConfiguration(S3Configuration.builder().useArnRegionEnabled(true).build())
                 .credentialsProvider(credentialsProvider).region(r).build());
     }
 
     /**
-     * Set s3 endpoint type.
+     * Update system property and refresh client.
+     */
+    private void handleSystemPropertyUpdate() {
+        // handle s3 endpoint type
+        boolean refreshCache = handleS3EndpointType(Coerce.toString(deviceConfiguration.gets3EndpointType()));
+
+        // handle fips endpoint, check current useFips value is consistent with deviceConfig fipsMode value,
+        // refresh client if they are inconsistent
+        if (useFips != Coerce.toBoolean(deviceConfiguration.getFipsMode())) {
```

Comment:
You should be checking fips system property with the mode in device config. 

Nit: If the s3 client is global and L106 is false, we can return as fips mode is not supported by s3 global endpoint. But, I think it's ok to refresh the client like this.

Reply:
Add lock for useFips.

Reply:
or use atomic compareAndSet() as the value obtained for `deviceConfiguration.getFipsMode()` in L110 may not be the same as L112.

Reply:
Updated